app/experimenter/experiments/serializers.py

```diff
@@ -508,6 +507,16 @@ def validate(self, data):
 
         return data
 
+    def is_variant_valid(self, variants):
+        unique_names = len(
+            set([slugify(variant["name"]) for variant in variants])
+        ) == len(variants)
+        all_contains_alphanumeric_and_spaces = all(
+            [re.match("^[A-Za-z0-9 ]*$", variant["name"]) for variant in variants]
+        )
```

Comment:
There are a couple very small improvements you could make here. First, I'd remove the square brackets `[]`. With those in place, it will create a intermediate list and then process the list. Without the square brackets you have a generator expression that will stream the data to the `all` method without creating an intermediate result.

Second, I'd pull the regex creation out of the loop, so that you aren't re-compiling the regex for each variant. Something like

```
alpha_and_spaces_re = re.compile(r"^[a-z0-9 ]+$", re.IGNORE_CASE)
all_contains_alpha_numeric_and_spaces = all(
  alpha_and_spaces_re.match(variant["name"]) for variant in variants
)
```

It may also be worth pulling that regex out into a class or module variable so it can be computed once instead of once per function call.